src/diagnostics.rs

```diff
@@ -2,27 +2,27 @@
 //! 
 //! The following data types are copied from [rust-lang/rust](https://github.com/rust-lang/rust/blob/de78655bca47cac8e783dbb563e7e5c25c1fae40/src/libsyntax/json.rs)
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Deserialize, Debug)]
 pub struct Diagnostic {
     /// The primary error message.
     pub message: String,
-    pub code: Option<DiagnosticCode>,
+    code: Option<DiagnosticCode>,
     /// "error: internal compiler error", "error", "warning", "note", "help".
-    pub level: String,
+    level: String,
     pub spans: Vec<DiagnosticSpan>,
     /// Associated diagnostic messages.
     pub children: Vec<Diagnostic>,
     /// The message as rustc would render it. Currently this is only
     /// `Some` for "suggestions", but eventually it will include all
     /// snippets.
-    pub rendered: Option<String>,
+    rendered: Option<String>,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Deserialize, Debug)]
 pub struct DiagnosticSpan {
     pub file_name: String,
-    pub byte_start: usize,
-    pub byte_end: usize,
+    byte_start: u32,
+    byte_end: u32,
```

Comment:
That's fair. We are not gonna run rustfix on code generated by lalrpop after all